jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1523,6 +1539,19 @@ public void onCompleted()
         }
     }
     
+    /**
+     * Called when a response is about to be committed, ie sent
+     * back to the client
+     */
+    public void onResponseCommit()
+    {
+        if (_sessions != null && _sessions.size() > 0)
```

Comment:
probably don't need the explicit `size()>0` check as the JVM will efficiently handle that case anyway for iterations over empty `ArrayList`s